### PR TITLE
Trie in Python

### LINKED ISSUE #839: Trie in Python
## 🚀 Feature

(A clear and concise description of what the feature is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/smv1999/CompetitiveProgrammingQuestionBank/blob/master/CONTRIBUTING.md)?

Yes

## Motivation

(Please outline the motivation for the proposal.)

## Pitch

It code was there in cpp language but I want to add a Python version of the code

### PR DESCRIPTION
**Pull request by PrashantVIT1**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/smv1999/CompetitiveProgrammingQuestionBank/blob/master/CONTRIBUTING.md)?

Yes
### Description

I have implemented Trie using dictionary(adding new word & search of added words)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.

## Related Issues or Pull Requests

closes #839

## Fill up the DCP form 

If you are a part of DCP 21, fill up this [form](https://docs.google.com/forms/d/e/1FAIpQLSeRAJnHE4eAa_W0IUeNZgheiy5POX3I3tQB7DtFsiB13D5w2g/viewform) after the PR is merged.

### The links provided in the form will be checked thoroughly, if someone tries fooling around, strict actions will be taken against the individual.

### CODE REVIEW

**Review by PrashantVIT1 — COMMENTED** [review]
Sorry for the inconvenience I am new to GitHub so I had no idea how uploading new file will affect. I have just added the 2nd code now that closes #853 
please inform me something has to be done

**Review by smv1999 — CHANGES_REQUESTED** [review]
Submit the solution in the format of .py file. Also, remove the extra file that is related to this PR.

**Comment by PrashantVIT1** [comment]
ok

**Comment by PrashantVIT1** [comment]
changes done

**Review by noviicee — CHANGES_REQUESTED** [review]
Add some comments to explain your code logic. Take dynamic input from the user.

**Comment by PrashantVIT1** [comment]
ok

**Review by PrashantVIT1 — COMMENTED** [review]
done comments and dynamic input

**Review by PrashantVIT1 — COMMENTED** [review]
upload in .py file format done

File: Trie/Trie_add_Search_using_dict.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+"""
```
Comment (noviicee) [inline Trie/Trie_add_Search_using_dict.py:1]: Just add a line here which explains the overall purpose of the code.

File: Trie/Trie_add_Search_using_dict.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+for j in range(n3):
+    checks.append(input())
+
+        
+dictionary = Trie()
+
+for k in words:
+    dictionary.add(k)
+for k in checks:
+    print(dictionary.search(k))
```
Comment (noviicee) [inline Trie/Trie_add_Search_using_dict.py:51]: Add the space-time complexity too!

**Review by noviicee — CHANGES_REQUESTED** [review]

File: Trie/Trie_add_Search_using_dict.py (hunk 2 above)
Comment (PrashantVIT1) [reply Trie/Trie_add_Search_using_dict.py:51]: Time complexity :
Insert: O(word Length)
Search: O(word Length)

Comment (PrashantVIT1) [reply Trie/Trie_add_Search_using_dict.py:51]: Space complexity <= sum of length of words ()
*if there is no word with common prefix is stored then Space complexity O(sum of length of words)
*if each word is prefix of some other word Space complexity will be O(largest word length)

File: Trie/Trie_add_Search_using_dict.py (hunk 1 above)
Comment (PrashantVIT1) [reply Trie/Trie_add_Search_using_dict.py:1]: Purpose:
* Implementation of Trie has been asked in my interviews like in Google 
* Store word in Dictionary 
* Search word in the dictionary returns True if exist
* Stores words in a compressed manner

**Review by PrashantVIT1 — COMMENTED** [review]
All the requested changes has been made. Now reply asap it's last day if any other changes are required.

**Review by noviicee — APPROVED** [review]

**Comment by PrashantVIT1** [comment]
Thank you this is my first merged PR

**Pull request merged**